### ISSUE TITLE
Add index to long help pages

### ISSUE THREAD

**Issue by garethrees** [mysociety/whatdotheyknow-theme#878]
Some of our help pages are quite long, making it difficult to find the advice that you're after. In the absence of being able to search help pages (https://github.com/mysociety/alaveteli/issues/233), we could at least make a quick improvement and add an index of major headings to the top of the page like we do on the the [pro help page](https://www.whatdotheyknow.com/help/pro).

![Screenshot 2021-08-25 at 14 27 43](https://user-images.githubusercontent.com/282788/130799770-039a2acc-4d45-437c-b2e7-c15103f6f522.png)

In terms of implementation, we'd just create a link to the anchor of each heading.

There'd be a little bit of maintenance involved in making sure we keep the index and headings in sync if we change the wording, but that doesn't sound particularly arduous and isn't something we do that often.

**Comment by mdeuk** [comment]
Linking to #883 as these are somewhat complimentary, yet distinct, issues.

**Comment by RichardTaylor** [comment]
I was going to write a ticket for "add ability to search the help pages" but that's probably impractical and better indexing would help. 

I often spend too much time looking for a help page section which I know, or suspect, is there somewhere.

**Comment by WilliamWDTK** [comment]
I've sort of created my own index with a bookmarks folder in my browser's toolbar, although I could only do it for (parts of) pages with anchors (#1140):

# Useful WDTK help links

## About
[About - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/about)
[Reporting Requests](https://www.whatdotheyknow.com/help/about#reporting)

---

## Making requests
[Making requests - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/requesting)
[Including bodies not covered by FOI/EIR](https://www.whatdotheyknow.com/help/requesting#authorities)
[How to write requests](https://www.whatdotheyknow.com/help/requesting#focused)
[Problems with/lack of responses](https://www.whatdotheyknow.com/help/requesting#no_response)
[SAR Advice](https://www.whatdotheyknow.com/help/requesting#data_protection)
[Why only EIR?](https://www.whatdotheyknow.com/help/requesting#eir)
[Upload request to site](https://www.whatdotheyknow.com/help/requesting#offsite)
[Add attachment to request/message](https://www.whatdotheyknow.com/help/requesting#attachments)
[Whistleblowers](https://www.whatdotheyknow.com/help/requesting#whistleblowers)

---

## Privacy
[Your privacy - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/privacy)
[Postal addresses / responses through the post](https://www.whatdotheyknow.com/help/privacy#full_address)
[Names](https://www.whatdotheyknow.com/help/privacy#Names)
[Pseudonyms](https://www.whatdotheyknow.com/help/privacy#real_name)
[GDPR Rights](https://www.whatdotheyknow.com/help/privacy#your_rights)
[Email address (and disclosure thereof)](https://www.whatdotheyknow.com/help/privacy#email_address)

---

## For FOI officers
[FOI officer questions - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/officers)
[Deadline calculation](https://www.whatdotheyknow.com/help/officers#days)
[Extensions](https://www.whatdotheyknow.com/help/officers#public_interest_test)
[Why "Awaiting classification"/"delayed"](https://www.whatdotheyknow.com/help/officers#classifications)
[Sending large files](https://www.whatdotheyknow.com/help/officers#large_file)
[Why publish names/email contents](https://www.whatdotheyknow.com/help/officers#names)
[Bounced responses](https://www.whatdotheyknow.com/help/officers#allowed_responses)

---

[Guidance for ICO and OSIC officers - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/ico_officers)
[Why we do this - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/principles)

---

## How WDTK is run
[How we run WhatDoTheyKnow - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/how)
[Reactive moderations](https://www.whatdotheyknow.com/help/how#reactive_moderation)
[Procedure after report](https://www.whatdotheyknow.com/help/how#what_happens)
[Removal of information](https://www.whatdotheyknow.com/help/how#information_removal)
[Vexatious requests](https://www.whatdotheyknow.com/help/how#vexatious)
[Annotation Moderation](https://www.whatdotheyknow.com/help/requesting#moderation)
[Invalid requests](https://www.whatdotheyknow.com/help/how#invalid_requests)
[Email/mobile redaction](https://www.whatdotheyknow.com/help/how#email_addresses)
[Admin team](https://www.whatdotheyknow.com/help/how#admin_team)
[User accounts](https://www.whatdotheyknow.com/help/how#user_accounts)
[Removal requests](https://www.whatdotheyknow.com/help/how#anonymise_account)
[Banning users](https://www.whatdotheyknow.com/help/how#banning_users)

---

[House rules - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/house_rules)
[Complaints about us - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/complaints)
[Get involved - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/volunteers)
[Beginner’s Guide - WhatDoTheyKnow](https://www.whatdotheyknow.com/help/beginners)

---

# WDTK Pro
[Pro](https://www.whatdotheyknow.com/pro)
[Pro help](https://www.whatdotheyknow.com/help/pro)
[Pro / Embargoed Requests](https://www.whatdotheyknow.com/help/requesting#private_requests)
[Batch requests](https://www.whatdotheyknow.com/help/requesting#multiple)

**Comment by WilliamWDTK** [comment]
> I was going to write a ticket for "add ability to search the help pages" but that's probably impractical and better indexing would help.
> 
> I often spend too much time looking for a help page section which I know, or suspect, is there somewhere.

There already is one:

mysociety/alaveteli#233

**Comment by sallytay** [comment]
I agree a search for help pages would be really useful, if we struggle to find things sometimes then users will too, and this could reduce the number of emails that we get for help and advice when the information is already available on site.

Your suggestion for improved indexing would definitely help with this, and I think would be easier to implement on the site.

Thank you for putting this together.
